src/ring_theory/valuation/basic.lean

```diff
@@ -0,0 +1,392 @@
+    replace hy := v.ne_zero_iff.mpr hy,
+    replace H := le_of_le_mul_right hy H,
+    rwa h at H, },
+  { apply mul_le_mul_right',
+    replace hy := v'.ne_zero_iff.mpr hy,
+    replace H := le_of_le_mul_right hy H,
+    rwa h, },
+end
+
+section supp
+variables  [comm_ring R]
+variables (v : valuation R Γ₀)
+
+/-- The support of a valuation v : R → Γ₀ is the ideal of R where v vanishes. -/
+def supp : ideal R :=
+{ carrier   := {x | v x = 0},
+  zero_mem' := map_zero v,
+  add_mem'  := λ x y hx hy, le_zero_iff.mp $
+    calc v (x + y) ≤ max (v x) (v y) : v.map_add x y
+               ... ≤ 0               : max_le (le_zero_iff.mpr hx) (le_zero_iff.mpr hy),
+  smul_mem' := λ c x hx, calc v (c * x)
+                        = v c * v x : map_mul v c x
+                    ... = v c * 0 : congr_arg _ hx
+                    ... = 0 : mul_zero _ }
+
+@[simp] lemma mem_supp_iff (x : R) : x ∈ supp v ↔ v x = 0 := iff.rfl
+-- @[simp] lemma mem_supp_iff' (x : R) : x ∈ (supp v : set R) ↔ v x = 0 := iff.rfl
+
+/-- The support of a valuation is a prime ideal. -/
+instance : ideal.is_prime (supp v) :=
+⟨λ (h : v.supp = ⊤), one_ne_zero $ show (1 : Γ₀) = 0,
+from calc 1 = v 1 : v.map_one.symm
+        ... = 0   : show (1:R) ∈ supp v, by rw h; trivial,
+ λ x y hxy, begin
+    show v x = 0 ∨ v y = 0,
+    change v (x * y) = 0 at hxy,
+    rw [v.map_mul x y] at hxy,
+    exact eq_zero_or_eq_zero_of_mul_eq_zero hxy
+  end⟩
+
+/-- v(a)=v(a+s) if s ∈ supp(v). -/
+lemma val_add_supp (a s : R) (h : s ∈ supp v) : v (a + s) = v a :=
+begin
+  have aux : ∀ a s, v s = 0 → v (a + s) ≤ v a,
+  { intros a' s' h', refine le_trans (v.map_add a' s') (max_le (le_refl _) _), simp [h'], },
+  apply le_antisymm (aux a s h),
+  calc v a = v (a + s + -s) : by simp
+       ... ≤ v (a + s)      : aux (a + s) (-s) (by rwa ←ideal.neg_mem_iff at h)
+end
+
+/-- If `hJ : J ⊆ supp v` then `on_quot_val hJ` is the induced function on R/J as a function.
+Note: it's just the function; the valuation is `on_quot hJ`. -/
+definition on_quot_val {J : ideal R} (hJ : J ≤ supp v) :
+  J.quotient → Γ₀ :=
+λ q, quotient.lift_on' q v $ λ a b h,
+calc v a = v (b + (a - b)) : by simp
+     ... = v b             : v.val_add_supp b (a - b) (hJ h)
+
+/-- The extension of valuation v on R to valuation on R/J if J ⊆ supp v -/
+definition on_quot {J : ideal R} (hJ : J ≤ supp v) :
+  valuation J.quotient Γ₀ :=
+{ to_fun := v.on_quot_val hJ,
+  map_zero' := v.map_zero,
+  map_one'  := v.map_one,
+  map_mul'  := λ xbar ybar, quotient.ind₂' v.map_mul xbar ybar,
+  map_add'  := λ xbar ybar, quotient.ind₂' v.map_add xbar ybar }
+
+@[simp] lemma on_quot_comap_eq {J : ideal R} (hJ : J ≤ supp v) :
+  (v.on_quot hJ).comap (ideal.quotient.mk_hom J) = v :=
+ext $ λ r,
+begin
+  refine @quotient.lift_on_beta _ _ (J.quotient_rel) v (λ a b h, _) _,
+  calc v a = v (b + (a - b)) : by simp
+       ... = v b             : v.val_add_supp b (a - b) (hJ h)
+end
+
+end supp -- end of section
+
+section supp_comm
```

Comment:
What's the difference between this section and the previous one? Didn't you already assume `R` was a `comm_ring`?

Reply:
No idea why this happened. Fixed.